src/components/volume-viewer/VolumeViewer.jsx

```diff
@@ -0,0 +1,106 @@
+import { Entity, Scene } from "aframe-react";
+import { Container } from "react-bootstrap";
+
+import "../../Aframe/arcball-camera";
+
+import {
+  volumePosition,
+  volumeRotation,
+  volumeScale,
+} from "../../assets/config.json";
+import Spinner from "./Spinner";
+
+import { useControlsContext } from "../../context/controls-context";
+
+function getCoordinates(transferFunctionNodes, plane) {
+  let coordinates = [];
+
+  if (plane === "x" || plane === "y") {
+    transferFunctionNodes.forEach((node) => {
+      coordinates.push(node[plane]);
+    });
+  } else console.error("Invalid Plane", plane);
+
+  return coordinates;
+}
+
+export default function VolumeViewer(props) {
+  const { volume } = props;
+  const { state: controlsState } = useControlsContext();
+
+  return (
+    <Container fluid className="aframe-container mb-3" id="visualizer">
+      <div id="modelLoaded" style={{ display: "none" }}>
+        <Spinner />
+      </div>
+
+      <Scene id="myScene" background="color: black" embedded>
+        <Entity
+          id="rhand"
+          laser-controls="hand: right"
+          raycaster="objects: .clickableMesh"
+          my-buttons-check={{ clipPlane: false, grabObject: false }}
+          collider-check={{ intersecting: false }}
+        />
+
+        <Entity
+          id="my2DclipplaneListener"
+          render-2d-clipplane={{
+            activateClipPlane: true,
+            xCLipPlaneMin: controlsState.xLowerBound,
+            xCLipPlaneMax: controlsState.xUpperBound,
+            yCLipPlaneMin: controlsState.yLowerBound,
+            yCLipPlaneMax: controlsState.yUpperBound,
+            zCLipPlaneMin: controlsState.zLowerBound,
+            zCLipPlaneMax: controlsState.zUpperBound,
+            currenAxisAngle: "0 0 0",
+            rotateAngle: "0 0 0",
+            clipX: "0 0",
+          }}
+        />
+        <a-plane
+          class="clickable"
+          id="my2Dclipplane"
+          visible="false"
+          height="1"
+          width="1"
+          material="color: red ; side:double; transparent:true;opacity:0.3"
+          cursor-listener
+        />
+
+        <Entity
+          id="volumeCube"
+          class="clickableMesh"
+          myloader={{
+            useTransferFunction: controlsState.USE_TRANSFER_FUNCTION,
```

Comment:
why is this state all caps when the rest is camelCase?

Reply:
uppercase is typical for global constants, but in this case while we know it is constant, it comes from the config and could be changed, so I'd keep it in line with other state variable names. @broarr what's your take?

Reply:
The react convention is to make all states regardless of where they come from `TRAIN_CASE`, I think I'd follow that here too